src/main/java/chess/Piece.java

```diff
@@ -58,32 +64,29 @@ public boolean move(final Position target) {
             if (chessBoard.getBoard()[target.getColumn()][target.getRow()] != null) {
                 chessBoard.getBoard()[target.getColumn()][target.getRow()].die();
             }
-            chessBoard.addPiece(this);
             this.position = target;
-            chessBoard.getBoard()[previousPosition.getColumn()][previousPosition.getRow()] = null;
-            if (this instanceof King && (this.getPosition().getColumn() - previousPosition.getColumn() == CASTLING_LEFT || this.getPosition().getColumn() - previousPosition.getColumn() == CASTLING_RIGHT)) {
-                Piece rookMoved;
-                int targetCol;
-                if (this.getPosition().getColumn() - previousPosition.getColumn() == CASTLING_RIGHT) {
-                    rookMoved = chessBoard.getBoard()[CASTLING_RIGHT_ROOK_SOURCE][previousPosition.getRow()];
-                    targetCol = CASTLING_RIGHT_ROOK_TARGET;
+            chessBoard.movePiece(this, previousPosition);
+            if (this instanceof King && abs(this.getPosition().getColumn() - previousPosition.getColumn()) == 2) {
```

Comment:
try to use a constant instead of use "2"

Reply:
Thanks, it has solved in last commit.